src/client/app/containers/CompareChartContainer.ts

```diff
@@ -41,107 +28,76 @@ function mapStateToProps(state: State, ownProps: CompareChartContainerProps): Li
 	const changeSummary = getCompareChangeSummary(entity.change, entity.name, periodLabels);
 
-	labels.push(periodLabels.prev);
-	labels.push(periodLabels.current);
 	const barColor = 'rgba(218, 165, 32, 1)';
+
+	const previousPeriod = entity.prevUsage;
+	const currentPeriod = entity.currUsage;
+
+
 	datasets.push(
 		{
-			data: [entity.prevUsage, entity.currUsage],
-			datalabels: {
-				anchor: 'end',
-				align: 'start'
+			x: [periodLabels.prev, periodLabels.current],
+			y: [previousPeriod, currentPeriod],
+			hovertext: [
+				`<b>${previousPeriod} KW</b> ${translate('used.this.time')}<br>${periodLabels.prev.toLowerCase()}`,
+				`<b>${currentPeriod} KW</b> ${translate('used.so.far')}<br>${periodLabels.current.toLowerCase()}`
+			],
+			hoverinfo: 'text',
+			type: 'bar',
+			marker: {color: barColor},
+			text: [ `<b>${previousPeriod} kW</b>`, `<b>${currentPeriod} kW</b>`],
+			textposition: 'auto',
+			textfont: {
+			  color: 'rgba(0,0,0,1)'
 			}
 		}
-	);
-	datasets[0].backgroundColor = barColor;
-	datasets[0].hoverBackgroundColor = barColor;
+	)
 
-	const data: ChartData = {datasets, labels};
-	const ticks: LinearTickOptions = {
-		beginAtZero: true
-	};
-	const options = {
-		animation: {
-			duration: 0
-		},
-		elements: {
-			point: {
-				radius: 0
-			}
-		},
-		scales: {
-			xAxes: [{
-				stacked: true,
-				gridLines: {
-					display: true
-				}
-			}],
-			yAxes: [{
-				stacked: false,
-				scaleLabel: {
-					display: true,
-					labelString: 'kW'
-				},
-				ticks
-			}]
+	const layout: any = {
+		title: `<b>${changeSummary}</b>`,
+		titlefont: {
+			size: 10,
+			color: colorize(entity.change)
 		},
+		hovermode:'closest',
+		autozise: true,
+		showlegend: false,
 		legend: {
-			display: false
 		},
-		tooltips: {
-			mode: 'nearest',
-			intersect: false,
-			backgroundColor: 'rgba(0,0,0,0.6)',
-			displayColors: false,
-			callbacks: {
-				label: (tooltipItem: ChartTooltipItem, data: ChartData) => { // tslint:disable-line no-shadowed-variable
-					const usage = tooltipItem.yLabel;
-					const usedThisTime = data.datasets![0].data![0];
-					const usedSoFar = data.datasets![0].data![1];
-					const labelText = tooltipItem.xLabel!.toLowerCase();
-					switch (usage) {
-						case usedThisTime:
-							return `${usage} kW ${translate('used.this.time')} ${labelText}`;
-						case usedSoFar:
-							return `${usage} kW ${translate('used.so.far')} ${labelText}`;
-						default:
-							return '';
-					}
-				},
-				title: () => ''
-			}
+		yaxis: {
+			title: 'kW',
+			showgrid: true,
+			gridcolor: '#ddd',
 		},
-		title: {
-			display: true,
-			text: changeSummary,
-			fontColor: colorize(entity.change)
+		xaxis: {
+			showgrid: false,
+			gridcolor: '#ddd'
 		},
-		plugins: {
-			datalabels: {
-				color: 'black',
-				font: {
-					weight: 'bold'
-				},
-				display: true,
-				formatter: (value: number) => `${value} kW`
-			}
+		margin: {
+			t: 20,
+			b: 120,
+			l: 60,
+			r: 20,
```

Comment:
In the Plotly reference at <https://plot.ly/javascript/reference/#layout-margin>, I cannot find the r attribute for margin. Want to verify this is correct usage.

Reply:
I noted a similar issue but found that the attribute is seen in the search function of plotly's documentation although not found in the documentation itself. I have put in an issue on Plotly's Documentation GitHub regarding this.

Reply:
Let's assume it is okay to use, esp. since not causing any issues.